Update article ruleset to v25.

biancadanforth opened the pull request:
Edit: Turns out I just had a transcription error in copying my ruleset over into the demo -- I was missing some paragraph rule coefficients.

--

Unfortunately, I am not seeing the 9/10 kind of performance I would expect when randomly visiting pages given my [test accuracy per tag of 93%](https://github.com/mozilla-services/fathom-smoot/commit/25047c6569a4480625579eff32dba4e257ba75e2)[1].

This indicates to me that the random collection of samples Vlad put together is not necessarily representative.

I could try:
* These same pages on `smoot-demo` (without the revised ruleset). Does it perform better?
* These pages in mozilla-services/fathom-smoot (outside of the Smoot demo project) with my current article v25 ruleset and see if they have the same confidences.

[1]:
Some example sites that are failing consistently:
  * Fast Company:
    * 0%: https://www.fastcompany.com/90452018/climate-change-wont-result-in-a-new-normal-but-in-constant-horrifying-new-disasters
  * Wikipedia
    * 6%: https://en.wikipedia.org/wiki/Banana
    * 100%: https://en.wikipedia.org/wiki/Trousers
    * 6%: https://en.wikipedia.org/wiki/Elephant
  * Npr.org
    * 0%: https://www.npr.org/2020/01/16/796994699/aussie-firefighters-save-worlds-only-groves-of-prehistoric-wollemi-pines
    * 0%: https://www.npr.org/2020/01/16/796328145/electric-burn-those-who-bet-against-elon-musk-and-tesla-are-paying-a-big-price
    * 0%: https://www.npr.org/sections/health-shots/2020/01/16/796316186/scientists-sent-mighty-mice-to-space-to-improve-treatments-back-on-earth
  * Cnn.com
    * 0%: https://www.cnn.com/2020/01/16/politics/pompeo-state-department-silence-new-evidence/index.html
    * 0%: https://www.cnn.com/2020/01/16/politics/saudi-arabia-us-troops-payment/index.html
    * 0%: https://www.cnn.com/2020/01/15/tech/ai-job-interview/index.html

biancadanforth committed:
Update article ruleset to v25.

Unfortunately, I am not seeing the 9/10 kind of performance I would expect when randomly visiting pages given my [test accuracy per tag of 93%](https://github.com/mozilla-services/fathom-smoot/commit/25047c6569a4480625579eff32dba4e257ba75e2).

This indicates to me that the random collection of samples Vlad put together is not necessarily representative.

I could try:
* These same pages on `smoot-demo` (without the revised ruleset). Does it perform better?
* These pages in mozilla-services/fathom-smoot (outside of the Smoot demo project) with my current article v25 ruleset and see if they have the same confidences.

Some example sites that are failing consistently:
  * Fast Company:
    * 0%: https://www.fastcompany.com/90452018/climate-change-wont-result-in-a-new-normal-but-in-constant-horrifying-new-disasters
  * Wikipedia
    * 6%: https://en.wikipedia.org/wiki/Banana
    * 100%: https://en.wikipedia.org/wiki/Trousers
    * 6%: https://en.wikipedia.org/wiki/Elephant
  * Npr.org
    * 0%: https://www.npr.org/2020/01/16/796994699/aussie-firefighters-save-worlds-only-groves-of-prehistoric-wollemi-pines
    * 0%: https://www.npr.org/2020/01/16/796328145/electric-burn-those-who-bet-against-elon-musk-and-tesla-are-paying-a-big-price
    * 0%: https://www.npr.org/sections/health-shots/2020/01/16/796316186/scientists-sent-mighty-mice-to-space-to-improve-treatments-back-on-earth
  * Cnn.com
    * 0%: https://www.cnn.com/2020/01/16/politics/pompeo-state-department-silence-new-evidence/index.html
    * 0%: https://www.cnn.com/2020/01/16/politics/saudi-arabia-us-troops-payment/index.html
    * 0%: https://www.cnn.com/2020/01/15/tech/ai-job-interview/index.html

biancadanforth committed:
Handle case when Fathom finds no candidate nodes.

For some pages, we would get a 'TypeError' if e.g. 'shoppingRules.against(this.document).get(shopping)' returned 0 elements.

biancadanforth committed:
Add missing 'paragraph' rule coefficients.

When I was manually copying over the ruleset, I must have made an error, as some rule coefficients were missing.

erikrose reviewed (approved):
Now matches fathom-smoot repo's performance. In it goes! :-)

biancadanforth committed:
Increase z-index to keep label on top.

Pages like [this one](https://www.cnbc.com/2020/01/27/airbnb-sabbatical-five-people-wanted-for-free-vacation-in-the-bahamas.html) were obscuring the label otherwise.
